pkg/comp-functions/functions/vshnredis/script/recreate.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -eo pipefail
+
+name="$STS_NAME"
+namespace="$STS_NAMESPACE"
+size="$STS_SIZE"
+release="$COMPOSITION_NAME"
+
+# Check if delete is necessary
+found=$(kubectl -n "$namespace" get sts "$name" -o json --ignore-not-found)
+
+foundsize=$(echo -En "$found" | jq -r '.spec.volumeClaimTemplates[] | select(.metadata.name=="redis-data") | .spec.resources.requests.storage')
+
+if [[ $foundsize != "$size" ]]; then
+  kubectl -n "$namespace" delete sts "$name" --cascade=orphan --ignore-not-found --wait=true
+  # Poke the release so it tries again to create the sts
+  # We first set it to garbage to ensure that the release is in an invalid state, we use an invalid state so it doesn't
+  # actually deploy anything.
+  # Then we patch the right size to enforce an upgrade
+  # This is necessary as provider-helm doesn't actually retry failed helm deployments unless the values change.
+  kubectl patch release "$release" --type merge -p "{\"spec\":{\"forProvider\":{\"values\":{\"master\":{\"persistence\":{\"size\":\"foo\"}}}}}}"
+  kubectl patch release "$release" --type merge -p "{\"spec\":{\"forProvider\":{\"values\":{\"master\":{\"persistence\":{\"size\":\"$size\"}}}}}}"
+  while ! kubectl -n "$namespace" get sts "$name"; do
+    echo "waiting for sts to re-appear"
+    sleep 1
+  done
+  kubectl -n "$namespace" label sts "$name" --overwrite "sts-resize.vshn.net/resize-inplace=true"
+fi
```

Comment:
this script is kind of a hell, few remarks:
- can You add flag for debugging or just enable some `echo` statements to help engineers debug issues?
- can we add return code checks? currently this script is just yolo running some commands, `set -e` would just terminate, we rather prefer to know precisely where it failed, maybe add `x` flag?
- can we add logic to exit `while` loop instead of spamming our apis each second indefinitely? 

maybe rewrite it to other scripting language? python or something? why is it done in script? it looks pretty much like something that can be done in comp-function

Reply:
I tried to do it in a comp function and I ended up in race condition hell. So no this can't be done easily with comp functions. Also with comp functions this would span over multiple reconcile loops which would result in the whole process taking multiple minutes.

We're using shell scripts for such jobs in various places, for example redis restore and backup. If we insist on implementing this in a proper language then we should do it in go as well.

I'll implement your suggestions for the script.